tools/flash_shellys.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+# 
+#  Copyright (c) 2020 Andrew Blackburn
+#  All rights reserved
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+# 
+#      http://www.apache.org/licenses/LICENSE-2.0
+#  
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#  
+#  This script will probe for any shelly device on the network and it will
+#  attemp to update them to the lastest firmware version availible.
+#  This script will not flash any firmware to a device that is not already on a
+#  version of this firmware, if you are looking to flash your device from stock
+#  or any other firmware please follow instructions here: 
+#  https://github.com/mongoose-os-apps/shelly-homekit/blob/master/README.md
+#  
+#  usage: . flash_shellys.sh
+
+if [ "$(which brew 2>/dev/null)" == "" ]; then
+  while true; do
+    read -p "brew is not installed, would you like to install it ?" yn
+    case $yn in
+        [Yy]* ) install_brew; break;;
+        [Nn]* ) echo "brew is required for this script to fuction. now exiting..."; exit;;
+        * ) echo "Please answer yes or no.";;
+    esac
+  done
+fi
+
+function install_brew {
+  echo -e '\033[1mInstalling brew...\033[0m'
+  echo -e '\033[1mPlease follow instructions...\033[0m'
+  echo -e '\033[1mYou will be asked for your password to install breww...\033[0m'
+  echo -e '\033[1m \033[0m'
+  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+}
+
+if [ "$(which timeout 2>/dev/null)" == "" ]; then
+  echo -e '\033[1mInstalling coreutils...\033[0m'
+  brew install coreutils
+fi
+
+if [ "$(which jq  2>/dev/null)" == "" ]; then
+  echo -e '\033[1mInstalling jq...\033[0m'
+  brew install jq
+fi
+
+function convert_to_integer {
+  echo "$@" | awk -F "." '{ printf("%03d%03d%03d", $1,$2,$3); }';
+}
+
+function download {
+  curl -L -o shelly-flash.zip "$@"
+}
+
+function flash {
+  download "$@"
+  if [ -f shelly-flash.zip ];then
+    while true; do
+      read -p "Are you sure you want to flash $device to firmware version $lfw ?" yn
+      case $yn in
+        [Yy]* )  echo "Now Flashing.."; curl -F file=shelly-flash.zip http://$device/update; break;;
+        [Nn]* ) echo "Skipping..";break;;
+        * ) echo "Please answer yes or no.";;
+      esac
+    done
+  fi
+  echo "waiting for $device to reboot"
+  sleep 10
+  if [ $(curl -qsS http://$device/rpc/Sys.GetInfo | jq -r .fw_version) == $lfw ];then
```

Comment:
use timeout in your calls to curl - say, 5 seconds (`-m 5`)
the only exception should be the call where your upload the firmware